### ISSUE TITLE
TypeDoc displays a superfluous type on number constants

### ISSUE THREAD

**Issue by Zamiell** [TypeStrong/typedoc#2717]
Consider the following code:

```ts
const SECOND_IN_MILLISECONDS = 1000;
```

TypeDoc will generate a page with the following:

![image](https://github.com/user-attachments/assets/4b5fb08a-4ee4-43cd-91a2-93aa1979ec92)

Here, the type of "1000" is superfluous/confusing/noisy/distracting. Thus, I propose that when TypeDoc determines that the type is exactly the same thing as the value, then it should omit printing out the type.

**Comment by Corso02** [comment]
Hi @Zamiell and @Gerrit0.
I managed to fix this issue. 
There were 2 tests that were failing right after I forked and cloned this repo. No other tests are failing after I implemented my solution. 

I used the same code that is mentioned in the description of this issue, and here is how the docs are generated with my solution in place. 
![image](https://github.com/user-attachments/assets/1ebedc5a-d05b-4ff6-ade4-e14bc1aec4e8)

Can I submit Pull Request?
